snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs

```diff
@@ -1,34 +1,37 @@
 ﻿//<SNIPPET1>
 using System;
 using System.Runtime.InteropServices;
+using System.Threading;
 
 class MainFunction
 {
     static void Main()
     {
-    Console.WriteLine("\nStringToGlobalAnsi\n");
+        Console.WriteLine("\nStringToGlobalAnsi\n");
 
-    // Create a managed string.
-    String  managedString = "I am a managed String";
-    Console.WriteLine("1) managedString = " + managedString );
+        // Create a managed string.
+        String  managedString = "I am a managed String";
+        Console.WriteLine("1) managedString = " + managedString);
 
-    // Marshal the managed string to unmanaged memory.
-    IntPtr stringPointer = (IntPtr)Marshal.StringToHGlobalAnsi(managedString);
-    Console.WriteLine("2) stringPointer = {0}", stringPointer );
+        // Marshal the managed string to unmanaged memory.
+        IntPtr stringPointer = (IntPtr)Marshal.StringToHGlobalAnsi(managedString);
+        Console.WriteLine("2) stringPointer = {0}", stringPointer);
 
-    // Get the string back from unmanaged memory
-    String RetrievedString = Marshal.PtrToStringAnsi( stringPointer);
-    Console.WriteLine("3) Retrieved from unmanaged memory = " + RetrievedString );
+        // Get the string back from unmanaged memory.
+        String RetrievedString = Marshal.PtrToStringAnsi(stringPointer);
+        Console.WriteLine("3) Retrieved from unmanaged memory = " + RetrievedString);
 
-    // Always free the unmanaged string.
-    Marshal.FreeHGlobal(stringPointer);
+        // Always free the unmanaged string.
+        Marshal.FreeHGlobal(stringPointer);
 
-    // IntPtr handle value is still the same:
-    Console.WriteLine("4) stringPointer = " + stringPointer );
+        // IntPtr handle value is still the same:
+        Console.WriteLine("4) stringPointer = " + stringPointer);
```

Comment:
I'm really confused as to how this code can compile?

Reply:
Sorry, I didn't get what you mean. Do you mean it was supposed to have compilation errors on incorrect formats/styles? Possibly a style config issue?

Reply:
hahaha no never mind. I didn't notice that the `"` was missing, so I thought the code was `Console.WriteLine("4") other-code` which would be invalid. lol

Reply:
All good, thanks